src/plugins/rv-packages/openassetio_resolver/openassetio_resolver.py

```diff
@@ -0,0 +1,106 @@
+"""
+Resolver for openassetio assets
+"""
+# *****************************************************************************
+# Copyright (c) 2023 Autodesk, Inc.
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# *****************************************************************************
+
+import os
+import sys
+from functools import partial
+
+from rv.rvtypes import MinorMode
+
+from openassetio.log import ConsoleLogger, SeverityFilter
+from openassetio.hostApi import HostInterface, ManagerFactory
+from openassetio.pluginSystem import PythonPluginSystemManagerImplementationFactory
+
+from openassetio_mediacreation.traits.content import LocatableContentTrait
+
+
+class OpenAssetIOPlugin(MinorMode):
+    class OpenRvHost(HostInterface):
+        def identifier(self):
+            return "io.aswf.openrv"
+
+        def displayName(self):
+            return "OpenRV"
+
+
+    def __init__(self):
+        super(OpenAssetIOPlugin, self).__init__()
+        self.init(
+            "openassetio_resolver",
+            [
+                (
+                    "incoming-source-path",
+                    self.incoming_source_path,
+                    "Resolve incoming reference if its an OpenAssetIO URI",
+                ),
+                (
+                    "openassetio-get-related-references",
+                    self.get_related_references,
+                    "Return related references if there are any",
+                ),
+            ],
+
+            None,
+        )
+        self.manager = None
+
+        if os.environ.get("OPENASSETIO_DEFAULT_CONFIG"):
+            logger = SeverityFilter(ConsoleLogger())
+            impl_factory = PythonPluginSystemManagerImplementationFactory(logger)
+            host_interface = self.OpenRvHost()
+
+            self.manager = ManagerFactory.defaultManagerForInterface(host_interface, impl_factory, logger)
+
+            self.context = self.manager.createContext()
+            self.context.access = self.context.Access.kRead
+
+    @staticmethod
+    def resolve_asset(event, _, data):
+        locatable = LocatableContentTrait(data)
+        if locatable.isValid():
+            event.setReturnContent(LocatableContentTrait(data).getLocation())
```

Comment:
The `LocatableContentTrait` defines that `location` is a URL, so in production scenarios, for files this would be `file:///url%20encoded%20path` (and also `https://` for siggraph), is RV happy with that?

Reply:
It will definitely work with `https://`. I suspect the url encoded file path may pose a problem though, but it could be decoded here.